back/engines/commercial/insights/app/controllers/insights/web_api/v1/categories_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module Insights
+  module WebApi::V1
+    class CategoriesController < ::ApplicationController
+      def show
+        render json: serialize(category)
+      end
+
+      def index
+        render json: serialize(categories)
+      end
+
+      def create
+        category = authorize(Insights::Category.new(create_params))
+        if category.save
+          render json: serialize(category), status: :created
+        else
+          render json: { errors: category.errors.details }, status: :unprocessable_entity
+        end
+      end
+
+      def update
+        if category.update(update_params)
+          render json: serialize(category)
+        else
+          render json: { errors: category.errors.details }, status: :unprocessable_entity
+        end
+      end
+
+      def destroy
+        status = category.destroy.destroyed? ? :ok : 500
+        head status
+      end
+
+      def destroy_all
+        authorize(Insights::Category)
+        categories.destroy_all
+        status = categories.count.zero? ? :ok : 500
+        head status
+      end
+
+      private
+
+      def categories
+        @categories ||= policy_scope(
+          View.includes(:categories).find(params.require(:view_id)).categories
+        )
+      end
+
+      def create_params
+        @create_params ||= params.require(:category)
+                                 .permit(:name)
+                                 .merge(view_id: params.require(:view_id))
+      end
+
+      def update_params
+        @update_params ||= params.require(:category).permit(:name)
+      end
+
+      # @param categories One or a collection of categories
+      def serialize(categories)
+        Insights::WebApi::V1::CategorySerializer.new(categories).serialized_json
+      end
+
+      def category
+        @category ||= authorize(
+          Category.find_by(id: params.require(:id), view_id: params.require(:view_id))
```

Comment:
I think this should be `find_by!` to make the 404 work?

Reply:
Well spotted!